node/lib/cmd/commit-shadow.js

```diff
@@ -90,6 +90,12 @@ exports.configureParser = function (parser) {
         defaultValue: false,
         help: "use timestamp of HEAD + 1 instead of current time",
     });
+    parser.addArgument(["-s", "--include-subrepos"], {
+        type: "string",
+        required: false,
+        defaultValue: "",
+        help: "only include specified sub-repos, includes all if left empty",
```

Comment:
No strong opinion on this but would like feedback from @novalis -- should empty-list really imply all? It seems to me that the natural behavior for this (because it would require no special cases or documentation) would be for the empty list to mean none.

Reply:
Well, it sure makes migration easier!

But I take your point that it's a little out-of-sync with the rest of git, so probably it's worth the slight hassle.

Reply:
I don't think it affects migration, does it? The default when this argument is omitted would be to include all submodules. I'm just talking about the behavior when you use this argument but provide no paths to it.

Reply:
I agree, empty array == all is a little confusing but it matches the convention used in getRepoStatus and the alternatives (using null or undefined) seem more kludgey, especially because they would just be converted to an empty list to be passed to getRepoStatus.  +1 to getting some feedback from @novalis

Reply:
Yes, I think it make sense for the `makeShadowCommitForRepo` method, but not necessarily for the CL API -- but I don't feel that strongly about it. If you and @novalis are in agreement that it's OK as is, it's fine with me.

Reply:
We could have it error out or return the sha for head if the flag is set with no values passed

Reply:
I think you can configure argparse to that it fails unless you have at least one arg. I like that solution.